Fix most of the reflection warnings

rmangi opened the pull request:
This PR adds reflection warnings and fixes most of them. We found that we got significantly better garbage collection performance using cbass under high load with these additions, ugly as they may be :)

jonwiggins committed:
Fix most of the reflection warnings

tolitius commented:
makes a lot of sense.

> we got significantly better garbage collection performance using cbass under high load

out of curiosity, do you have any performance numbers you can share?

rmangi commented:
No, when I realized that garbage collection was so problematic I added these warnings to cbass and to my application code at the same time. We've started doing this by default, it unfortunately makes a big difference.

I can say that we're sending about 500K increment ops a minute with cbass and it works great with G1GC. We added increment and batch increment functions as well which I'll submit a PR for after this.  I can also say that the performance of this is allowing us to replace a massive riak cluster with a much leaner hbase one 😈

tolitius commented:
> I can also say that the performance of this is allowing us to replace a massive riak cluster with a much leaner hbase one.

that's great :)
